jupyterhub/roles.py

```diff
@@ -99,6 +131,15 @@ def expand_subscopes(index):
     return expanded_scope
 
 
+def get_scopes_for(orm_object):
+    """Get the scopes for User/Service/Group/Token"""
+    scopes = get_subscopes(*orm_object.roles)
+    if 'self' in scopes:
```

Comment:
Should probably add a check that `self` is only valid for orm.User

```suggestion
    if 'self' in scopes:
        if not isinstance(orm_object, orm.User):
            raise ValueError(f"'self' scope only valid for Users, got {orm_object}")
```